tests/integration/test_endpoints.py

```diff
@@ -219,6 +221,109 @@ def test_user_no_jwt(client):
     assert response.status_code == 401
 
 
+@pytest.mark.parametrize('input', [
+    # full definition
+    {
+        'type': 'gdpr',
+        'category': 'newsletter',
+        'status': 'accepted',
+        'timestamp': datetime.now(timezone('UTC')).isoformat(),
+        'valid_until': datetime.now(timezone('UTC')).isoformat(),
+        'message': 'I consent to the ToS',
+        'source': 'pytest'
+    },
+    # minimal
+    {
+        'type': 'cookie',
+        'category': 'preferences',
+        'status': 'rejected',
+    }
+])
+def test_create_consent(client, session, tokens, input):
+    """Create a new consent."""
+    response = client.post("/consent", json=input, headers={
+        'Authorization': f"Bearer {tokens['write']}",
+    })
+    assert response.status_code == 201
+    consent_id = response.json['id']
+    assert Consent.query.filter(Consent.id == consent_id).count() == 1
+
+
+@pytest.mark.parametrize('input', [
+    # incorrect type
+    {
+        'type': 'gpdr',
+        'category': 'newsletter',
+        'status': 'accepted',
+    },
+    # incorrect category for 'cookie' consent
+    {
+        'type': 'cookie',
+        'category': 'fumctiomal',
+        'status': 'accepted',
+    },
+    # incorrect status
+    {
+        'type': 'cookie',
+        'category': 'strictly_necessary',
+        'status': 'akcepted',
+    },
+])
+def test_create_consent_fail(client, session, tokens, input):
+    """Create a new consent."""
+    response = client.post("/consent", json=input, headers={
+        'Authorization': f"Bearer {tokens['write']}",
+    })
+    assert response.status_code == 422
+
+
+def test_get_consent(app, client, session, models, tokens):
```

Comment:
This test is good but I would consider splitting it into 3:

* Test just request and expected response code
* Test expected functioning of the sql query grouping and selecting latest consent
* Test for correctness and unique combination of type and category for each of the user's consents

It will be more verbose but more clear what each test is doing. I imagine that each of those cases could fail independently of each other.

Reply:
I've split them, both endpoint tests and model tests in https://github.com/DD-DeCaF/iam/pull/25/commits/7a46cd44cf314846a920751b3fed503f5ad5da13